src/Type/Definition/ScalarType.php

```diff
@@ -36,7 +35,7 @@ class ScalarType implements TypeInterface, NamedTypeInterface, LeafTypeInterface
      *
      * @param string                        $name
      * @param null|string                   $description
-     * @param callable|null                 $serializeCallback
+     * @param callable                      $serializeCallback
```

Comment:
After moving the invariant checks outside of the this class we can now ensure that the serialize function is provided.